specutils/wcs/adapters/gwcs_adapter.py

```diff
@@ -79,7 +79,7 @@ def rest_wavelength(self):
     def with_spectral_unit(self, unit, rest_value=None, velocity_convention=None):
         """
         """
-        if isinstance(unit, u.Unit) and unit.is_equivalent(
```

Comment:
What failure are you seeing using `isinstance(unit, u.Unit)`?

Reply:
for example, `isinstance(u.m, u.Unit)` will return false.